tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: 'How to use CPM to generate an off-chain TypeScript SDK'
+description: "The tutorial aims to provide a comprehensive guide for developers interested in creating TypeScript SDKs for Neo smart contracts, making it easier for them to leverage CPM and successfully work with off-chain SDKs."
+author: coz.io
+tags: ["TS", "CPM", "SDK"]
+skill: BEGINNER
+sidebar: true
+---
+
+[CPM](https://github.com/CityOfZion/cpm) is a tool that can help you download smart contracts and generate on-chain and off-chain SDKs for multiple languages without the need to write them yourself and needing none or minimal changes.
+
+## 1. Initializing workspace
+We recommend using [vite](https://vitejs.dev/guide/) to create your project from a template. You just need to run `npm create vite@latest` and select the prompts to generate your project. After creating your project, it's time to start using CPM to automate the off-chain SDK generation.
+
+## 2. Using CPM 
+It's pretty easy to use CPM to add an off-chain SDK to your TypeScript project. After installing CPM, you can run it with a `cpm.yaml` or you could directly use the command prompt. In this tutorial, we will be using the easiest one to maintain.
+
+We need to generate a default `cpm.yaml` file by running `cpm init` on the terminal, it should create a file that is getting the [Props](https://github.com/CityOfZion/props) smart contract. While it is a useful contract, it's probably not the one you want, in this article, let's suppose we want to easily invoke functions from the [Neo Token contract](https://dora.coz.io/contract/neo3/mainnet/0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5).
```

Comment:
```suggestion
We need to generate a configuration file by running `cpm init` on the terminal. This creates a `cpm.yaml` file that by default gets the [Props](https://github.com/CityOfZion/props) smart contract. While it is a useful contract, it's probably not the one you want, in this article, let's suppose we want to easily invoke functions from the [Neo Token contract](https://dora.coz.io/contract/neo3/mainnet/0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5).
```

Slightly modified to use stronger wording on the creation. Instead of assuming it might fail to create the file, we assume it will always be successful.